SYCL/NonUniformGroups/masked_sub_group.cpp

```diff
@@ -0,0 +1,77 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// UNSUPPORTED: cuda || hip
+
+#include <sycl/sycl.hpp>
+namespace syclex = sycl::ext::oneapi::experimental;
+
+class TestKernel;
+
+int main() {
+  sycl::queue Q;
+  sycl::buffer<bool, 1> MatchBuf{sycl::range{32}};
+  sycl::buffer<bool, 1> LeaderBuf{sycl::range{32}};
+  sycl::buffer<uint32_t, 1> MaskBuf{sycl::range{32}};
+
+  const auto NDR = sycl::nd_range<1>{32, 32};
+  Q.submit([&](sycl::handler &CGH) {
+    sycl::accessor MatchAcc{MatchBuf, CGH, sycl::write_only};
+    sycl::accessor LeaderAcc{LeaderBuf, CGH, sycl::write_only};
+    sycl::accessor MaskAcc{MaskBuf, CGH, sycl::write_only};
+    const auto KernelFunc = [=](sycl::nd_item<1> item)
+        [[sycl::reqd_sub_group_size(32)]] {
```

Comment:
We should probably check that the device supports this sub-group size and skip the test if it doesn't.

Reply:
Fixed in https://github.com/intel/llvm-test-suite/pull/1574/commits/33c11f91aa3544197315640bf9180eff21085d94.